src/app/services/course.service.ts

```diff
@@ -9,34 +9,50 @@ export class CourseService {
 
   public courses: ICourse[];
 
-  constructor() {
-    this.courses = (courseList as any).default;
+  constructor(
+    private httpClient: HttpClient
+  ) { }
+
+  public getList(start = 0, count = 3, textFragment = ''): Promise<ICourse[]> {
+    return this.httpClient.get<ICourse[]>('http://localhost:3004/courses', {
+      params: new HttpParams()
+        .set('start', `${start}`)
+        .set('count', `${count}`)
+        .set('sort', 'date')
+        .set('textFragment', textFragment)
+    })
+    .toPromise()
+    .then((courses: ICourse[]) => {
+      this.courses = courses;
+      return this.courses;
+    })
   }
 
-  public getList(): ICourse[] {
-    return this.courses;
-  }
-
-  public createCourse(course: ICourse): void {
-    this.courses = this.courses.concat({...course});
+  public createCourse(course: ICourse): Promise<ICourse> {
+    return this.httpClient.post('http://localhost:3004/courses', course)
+      .toPromise()
+      .then((course: ICourse) => {
+        return course;
+      })
   }
 
   public getCourseById(id: string): ICourse {
-    return this.courses.find(course => course.id === id);
+    return this.courses.find(course => course.id.toString() === id);
   }
 
   public getCourseTitle(id: string): string {
-    let course =  this.courses.find(course => course.id === id);
-    return course.title;
+    let course =  this.courses.find(course => course.id.toString() === id);
+    return course.name;
   }
 
   public updateCourse(id: string, course: ICourse): void {
-    let courseFiltered = this.courses.filter(course => course.id !== id);
+    let courseFiltered = this.courses.filter(course => course.id.toString() !== id);
```

Comment:
id.toString() and param type is string. let's back to this moment in the next task